infobat/tests/test_http.py

```diff
@@ -0,0 +1,220 @@
+import os
+import os.path
+import datetime
+import tempfile
+import urllib
+
+from twisted.internet import reactor
+from twisted.internet import defer
+from twisted.internet import endpoints
+from twisted.web import client as webclient
+from twisted.web import http_headers
+from twisted.web.iweb import IBodyProducer
+from twisted.trial.unittest import TestCase as TrialTestCase
+from zope.interface import implementer
+
+from infobat.http import makeSite
+
+# TODO: Relocate templates inside infobat/ package
+_TEMPLATES_DIR = os.path.join(os.path.dirname(__file__), '../../web')
+
+
+class WebUITestCase(TrialTestCase):
+    def setUp(self):
+        self.client = webclient.Agent(reactor)
+
+    @defer.inlineCallbacks
+    def startWebUI(self, dbpool_fake):
+        self.site = makeSite(_TEMPLATES_DIR, dbpool_fake)
+        self.endpoint = endpoints.TCP4ServerEndpoint(reactor, 8888)
+        self.listeningPort = yield self.endpoint.listen(self.site)
+        self.addCleanup(self.listeningPort.stopListening)
```

Comment:
As long as you're adding TODOs, might as well TODO moving this to a fake HTTP client. I thought treq had one, maybe? It's not a priority for this PR of course.

Reply:
I like keeping this as more of an integration test, at least for now.